cmd/wallet-sdk-gomobile/oauth2/registerclientresponse.go

```diff
@@ -62,18 +64,106 @@ func (r *RegisterClientResponse) ClientSecretExpiresAt() (int, error) {
 	return *r.goAPIRegisterClientResponse.ClientSecretExpiresAt, nil
 }
 
-// HasClientMetadata indicates whether this RegisterClientResponse has client metadata.
-func (r *RegisterClientResponse) HasClientMetadata() bool {
-	return r.goAPIRegisterClientResponse.ClientMetadata != nil
+// RegisteredMetadata returns the RegisteredMetadata object, which can be used to determine what metadata was
```

Comment:
How the registeredMetadata is different from the clientMetadata ?

Reply:
It's the metadata that actually gets registered on the server - the spec states that the server may reject or replace the client metadata values. The client should know about this so they can react accordingly (if necessary). See here for more info: https://datatracker.ietf.org/doc/html/rfc7591#section-3.2.1

Reply:
So using the the existing name (client metadata) in the response is misleading, since it may not match the client metadata that was sent. Otherwise, the fields are the same (except for the issuer state field, but that will be removed soon as it was just a temporary workaround)